### ISSUE TITLE
GradualRolloutRandom with 0%

### ISSUE THREAD

**Issue by shadone** [Unleash/client-specification#1]
I am working on a new client implementation (for Swift) and while implementing tests from this client-specification repo I noticed a potential problem with GradualRolloutRandom strategy implementations.

https://github.com/Unleash/unleash-client-node/blob/master/src/strategy/gradual-rollout-random.ts#L12

```js
export class GradualRolloutRandomStrategy extends Strategy {
    constructor() {
        super('gradualRolloutRandom');
    }

    isEnabled(parameters: any, context: Context) {
        const percentage: number = Number(parameters.percentage);
        const random: number = Math.round(Math.random() * 100);
        return percentage >= random;
    }
}
```

Notice the "greater or equals" on the last line `return percentage >= random` - wouldn't that succeed for rollouts percentage set to 0 if random happens to return 0? Should it not be "percentage > random" ?

**Comment by ivarconr** [comment]
Yes you are correct. The java implementation actually do 'random number +1'. 

We should add a test for this scenario also.

**Comment by shadone** [comment]
Technically there is a test for it - https://github.com/Unleash/client-specification/blob/master/specifications/05-gradual-rollout-random-strategy.json#L20-L30 - but due to randomness being involved it does not always trigger.

**Comment by ivarconr** [comment]
Oh dear, you are right of course. If there ever was an easy way to control randomness across platforms? 

Might need to leave this to a unit-test i the node client for now.

**Comment by shadone** [comment]
Maybe the client-specification could be extended to "mock" the random number generator - add an ability to provide expected "random" number, e.g.

```
      {
        "name": "Feature.B5",
        "description": "Disabled toggle with 0% rollout",
        "enabled": true,
        "strategies": [
          {
            "name": "gradualRolloutRandom",
            "parameters": {
              "percentage": "0"
            },
            "testFixtures": {
              randomSequence: [0] // the sequence of fake "random" numbers that the strategy should be looping through
            }
          }
        ]
      },
```

**Comment by judomu** [comment]
Hey, is there already a swift client out there :) ?

**Comment by shadone** [comment]
@judomu hi, sorry for my late response. I've got OK to open source my Unleash client for iOS. So far I pushed it to my private repo: https://gitlab.com/shadone/unleash-client-swift I've also released it to CocoaPods.
It is used in production for about a year. However. It is not a complete implementation of the spec, the biggest part that is missing is caching toggles locally (for which there is incomplete implementation in a branch) and reporting metrics. PRs welcome :)
